src/components/Articles.jsx

```diff
@@ -1,22 +1,28 @@
 import React, { useEffect, useState } from "react";
-import { getArticles } from "../api";
+import { useParams } from "react-router-dom";
+import { getArticles, getArticlesByTopic } from "../api";
 import ArticleCard from "./Article_card";
 import "../style/Article.css";
 
 const ArticlesList = () => {
+  const { topic } = useParams();
   const [articles, setArticles] = useState([]);
   const [error, setError] = useState(null);
 
   useEffect(() => {
-    getArticles()
-      .then((data) => {
-        setArticles(data);
-      })
-      .catch((err) => {
-        setError(err.message);
-      });
-  }, []);
+    const fetchArticles = () => {
+      const fetchFunction = topic ? getArticlesByTopic(topic) : getArticles();
```

Comment:
If you used the axios params object, we could have one function (getArticles) do the job of getting all articles - or - getting articles by topic